crates/sui-types/src/messages_grpc.rs

```diff
@@ -181,3 +181,34 @@ pub struct SystemStateRequest {
     // This is needed to make gRPC happy.
     pub _unused: bool,
 }
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct HandleCertificateResponseV3 {
+    pub effects: SignedTransactionEffects,
+    pub events: Option<TransactionEvents>,
+    // Input objects will only be populated in the happy path
+    pub input_objects: Option<Vec<Object>>,
+    // Output objects will only be populated in the happy path
+    pub output_objects: Option<Vec<Object>>,
+    pub auxiliary_data: Option<Vec<u8>>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct HandleCertificateRequestV3 {
+    pub certificate: CertifiedTransaction,
+
+    pub include_events: bool,
```

Comment:
Yeah, i think they should be. Just like objects you don't need these from every validator since the effects themselves commit to an events digest so we only need a few validators to return this info and can save the remaining validators bandwidth.